addons/hr_expense/models/hr_expense.py

```diff
@@ -389,40 +428,80 @@ def action_view_sheet(self):
+
+            values = {
+                'default_company_id': self.company_id.id,
+                'default_employee_id': self[0].employee_id.id,
+                'default_name': expense_name,
+                'default_expense_line_ids': [Command.set(todo.ids)],
+                'default_state': 'draft',
+                'create': False
+            }
+            values = [values]
         else:
-            dates = todo.mapped('date')
-            min_date = format_date(self.env, min(dates))
-            max_date = format_date(self.env, max(dates))
-            expense_name = min_date if max_date == min_date else "%s - %s" % (min_date, max_date)
-
-        values = {
-            'default_company_id': self.company_id.id,
-            'default_employee_id': self[0].employee_id.id,
-            'default_name': expense_name,
-            'default_expense_line_ids': [Command.set(todo.ids)],
-            'default_state': 'draft',
-            'create': False
-        }
+            sheets = [self.filtered(lambda x: x.payment_mode == 'own_account'), self.filtered(lambda x: x.payment_mode == 'company_account')]
+            values = []
+            for todo in sheets:
+                if len(todo) == 1:
+                    expense_name = todo.name
+                else:
+                    dates = todo.mapped('date')
+                    min_date = format_date(self.env, min(dates))
+                    max_date = format_date(self.env, max(dates))
+                    expense_name = min_date if max_date == min_date else "%s - %s" % (min_date, max_date)
+
+                vals = {
+                    'company_id': self.company_id.id,
+                    'employee_id': self[0].employee_id.id,
+                    'name': expense_name,
+                    'expense_line_ids': [Command.set(todo.ids)],
+                    'state': 'draft',
+                }
+                values.append(vals)
         return values
 
     def action_submit_expenses(self):
+
         context_vals = self._get_default_expense_sheet_values()
-        return {
-            'name': _('New Expense Report'),
-            'type': 'ir.actions.act_window',
-            'view_mode': 'form',
-            'res_model': 'hr.expense.sheet',
-            'target': 'current',
-            'context': context_vals,
-        }
+        if len(context_vals) > 1:
+            sheets = self.env['hr.expense.sheet'].create(context_vals)
+            return {
+                'name': _('New Expense Reports'),
+                'type': 'ir.actions.act_window',
+                'view_mode': 'list,form',
+                'res_model': 'hr.expense.sheet',
+                'domain': [('id', 'in', sheets.ids)],
+                'context': self.env.context,
+            }
+        else:
+            return {
+                'name': _('New Expense Report'),
+                'type': 'ir.actions.act_window',
+                'view_mode': 'form',
```

Comment:
Did you try it ? I think res_id is mandatory here.

Reply:
This is not 'originally' my code, I just moved it around and that's why it is showing. But I did try and it worked before and works after moving as well :)